Source/mxm/GB_colscale.c

```diff
@@ -179,6 +178,17 @@ GrB_Info GB_colscale                // C = A*D, column scale with diagonal D
         // determine if the values are accessed
         //----------------------------------------------------------------------
 
+        // Do not dup A->p, A->h into C yet; if we use CUDA, we'll do it on the
+        // GPU
+        int64_t *tmp_Ap = A->p ;
+        int64_t *tmp_Ah = A->h ;
+        A->p = NULL ;
+        A->h = NULL ;
+        GB_OK (GB_dup_worker (&C, C_iso, A, false, ztype)) ;
```

Comment:
I hadn't considered this option for using GB_dup_worker but I see from my code that it can work. However, it might be cleaner to revise GB_dup_worker, and pass in flags that disable the copy of specific parts of the matrix, like "bool Ap_dup" or something, for each component.  I can pull in this PR for now, but a "// FIXME: revise this to use flags for each component" can be added here.

Reply:
That makes sense. It looks like this wouldn't be a huge change, I can make a follow-up PR for this.